torch/csrc/jit/passes/peephole.cpp

```diff
@@ -16,67 +16,59 @@ namespace torch { namespace jit {
 // TODO: Decide what kind of fixed point strategy we will have
 void PeepholeOptimize(Block * block) {
   for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
-    auto* n = *it;
+    auto* node = *it;
 
-    for (Block * sub_block : n->blocks()) {
+    for (Block * sub_block : node->blocks()) {
         PeepholeOptimize(sub_block);
     }
 
     // XXX: remember that if you want to simplify an expression by combining multiple nodes
     // into a different one, then you need to check that they all belong to the given block
-    switch (n->kind()) {
-      case aten::expand: {
-        // Eliminate redundant expand
-        if (!n->input()->isTensor()) break;
-        // the sizes are dynamic
-        if(n->inputs().size() != 1) break;
-        if (n->get<std::vector<int64_t>>(attr::size) == n->input()->type()->expect<TensorType>()->sizes()) {
-          n->output()->replaceAllUsesWith(n->input());
-          // Let DCE clean up any unused nodes at this point
+    if (node->matches("aten::expand(Tensor self, int[] size, *, int implicit) -> Tensor") &&
+        node->knows(attr::size)) {
+      // x.expand(x.size()) == x
+      if (auto input_type = node->getValue(attr::self)->type()->cast<TensorType>()) {
```

Comment:
I don't like `getValue` as a function name. It sounds too generic. Can we make it `node->input(attr::self)`? This follows the pattern of `input(0)` and makes it clear that we are looking up an input rather an output.

Reply:
Sure, makes sense. I'll change that